docs/scripts/buildApi.ts

```diff
@@ -41,6 +42,8 @@ function writePrettifiedFile(filename: string, data: string, prettierConfigPath:
     );
   }
 
+  data = data.replace(/\\r\\n/g, '\\n');
```

Comment:
@mbrookes needed to add this in order to fix the differences that are added on windows

Reply:
Perhaps add a comment regarding why it's needed?

Reply:
I tried also the following combinations, but they don't seem to work well
```
data = data.replace(/\\r\\n/g, getLineFeed(data));
data = data.replace(/\r?\n/g, getLineFeed(data));
```

Also I don't understand why I am seeing still docs:api changes errors, when no files are changed on the PR ( at least locally for me..) Probably I broke the scenario for everyone else now 😕  https://app.circleci.com/pipelines/github/mui-org/material-ui/27211/workflows/138b2a8a-4241-4228-9c32-45866c401faf/jobs/196749 

@eps1lon maybe you may have idea on how to fix this properly?

Reply:
Anyway, will revert this change and try to handle it later in a separate PR.

Reply:
Created https://github.com/mui-org/material-ui/issues/23577 for tracking